src/data/data.cc

```diff
@@ -933,7 +933,17 @@ void SparsePage::PushCSC(const SparsePage &batch) {
   self_offset = std::move(offset);
 }
 
+template uint64_t
+SparsePage::Push(const data::DenseAdapterBatch& batch, float missing, int nthread);
+template uint64_t
+SparsePage::Push(const data::CSRAdapterBatch& batch, float missing, int nthread);
+template uint64_t
+SparsePage::Push(const data::CSCAdapterBatch& batch, float missing, int nthread);
+template uint64_t
+SparsePage::Push(const data::DataTableAdapterBatch& batch, float missing, int nthread);
```

Comment:
How about other adapters? Like FileAdapter.

Reply:
The Intel compiler didn't appear to complain. Where can I find the full list of adapters?

Reply:
Adding `FileAdapterBatch`. I see only 5 adapter batch types in src/data/adapter.h.

Reply:
This should be all.